src/actions/cmsActions.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import { Predicates } from '@prismicio/client';
+import type { Dispatch, GetState } from 'reducers/rootReducer';
+import prismicClient from 'services/prismic';
+import { reportErrorLog } from 'utils/common';
+import { CMS_DATA_TYPES, DOCUMENT_TYPE } from 'constants/cmsConstants';
+import { SET_TUTORIAL_DATA } from 'constants/onboardingConstants';
+import type { CMSData } from 'models/CMSData';
+import { getTutorialDataObject, isValidTutorialData } from 'utils/cms';
+
+const {
+  ONBOARDING_SCREENS_FOR_NATIVES: NATIVES,
+  ONBOARDING_SCREENS_FOR_NEWBIES: NEWBIES,
+} = CMS_DATA_TYPES;
+
+export const getTutorialDataAction = () => async (dispatch: Dispatch, getState: GetState) => {
+  const { appSettings: { data: { hasSeenTutorial } } } = getState();
+  // no need to do anything if user has already completed tutorial
+  if (hasSeenTutorial) return null;
+  return prismicClient.query(Predicates.any(DOCUMENT_TYPE, [NATIVES, NEWBIES]))
+    .then((res: CMSData) => {
+      const tutorialData = getTutorialDataObject(res);
+      if (!isValidTutorialData(tutorialData)) return;
+      dispatch({
+        type: SET_TUTORIAL_DATA,
+        payload: tutorialData,
+      });
+    })
+    .catch(e => reportErrorLog(e));
```

Comment:
style, readability: Could we use `await` + `try/catch` instead of `then()` + `catch()`?